src/esp32/esp32_spi_master.c

```diff
@@ -264,7 +264,10 @@ static void esp32_spi_get_rx_data(spi_dev_t *dev, uint8_t *data, size_t skip,
       if (skip > 0) {
         skip--;
       } else {
-        data[i++] = byte;
+        if (NULL != data) {
```

Comment:
`var == value`, it's more natural.
short condition bodies can be put on the same line.

```
if (data != NULL)  data[i] = byte;
```

Reply:
YES! I just thought you followed that nasty rule...

Reply:
what made you think so? it's not used anywhere in mos or its core libraries.
i know where this thing comes from, and it may have made sense at the time, but now we have compilers to catch that kind of stuff (== vs = typos), and gcc does.

Reply:
Nothing,,. watching too many different libs in a short amount of time perhaps...

Reply:
ok. so, no, we're not following that doctrine over here :)